jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -906,7 +918,36 @@ public MethodHandle findStaticSetter(Class<?> clazz, String fieldName, Class<?>
 			/*[IF ]*/
 			/* If the new lookup class differs from the old one, protected members will not be accessible by virtue of inheritance. (Protected members may continue to be accessible because of package sharing.) */
 			/*[ENDIF]*/
+			/*[IF !Sidecar19-SE-B175]
 			int newAccessMode = accessMode & ~PROTECTED;
+			/*[ELSE]*/
+			/* The UNCONDITIONAL bit is discarded if the new lookup class differs from the old one in Java 9 */
+			int newAccessMode = accessMode & ~UNCONDITIONAL;
+			
+			/* There are 3 cases to be addressed for the new lookup class from a different module:
+			 * 1) There is no access if the package containing the new lookup class is not exported to 
+			 *    the package containing the old one.
+			 * 2) There is no access if the old lookup class is in a named module
+			 *    Note: The public access will be reserved if the old lookup class is a public lookup.
+			 * 3) The MODULE access is removed if the old lookup class is in an unnamed module.
+			 */
+			Module accessClassModule = accessClass.getModule();
+			Module lookupClassModule = lookupClass.getModule();
+			
+			if (!Objects.equals(accessClassModule, lookupClassModule)) {
+				if (!lookupClassModule.isExported(lookupClass.getPackageName(), accessClassModule)) {
```

Comment:
Where does this isExported check come from?  I don't see a reference to it in the conditions in the doc.

Reply:
Unfortunately this is not documented in the API Spec but it was detected in failing JCK tests.
This check is based on the explanation of Accessibility for the Module System at http://openjdk.java.net/projects/jigsaw/spec/sotms/

2.4  Accessibility
...
Thus, even when a type is declared public, if its package is not exported in the declaration of its module then it will only be accessible to code in that module.